substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2020,6 +2020,7 @@ pub mod pallet {
 			ensure!(!PoolMembers::<T>::contains_key(&who), Error::<T>::AccountBelongsToOtherPool);
 
 			let mut bonded_pool = BondedPool::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+			ensure!(bonded_pool.state == PoolState::Open, Error::<T>::NotOpen);
```

Comment:
eh, seems like a bug from the past? let's add a unit test.

Reply:
not sure why I did that but this is checked inside ok_to_join().